internal/services/datafactory/data_factory_resource.go

```diff
@@ -297,14 +308,20 @@ func resourceDataFactoryCreateUpdate(d *pluginsdk.ResourceData, meta interface{}
 			return fmt.Errorf("could not parse Key Vault Key ID: %+v", err)
 		}
 
-		identityIdsRaw := d.Get("identity.0.identity_ids").([]interface{})
+		userAssignedIdentityId := ""
+		if features.ThreePointOhBeta() {
+			userAssignedIdentityId = d.Get("customer_managed_key_identity_id").(string)
+		} else {
+			identityIdsRaw := d.Get("identity.0.identity_ids").([]interface{})
+			userAssignedIdentityId = identityIdsRaw[0].(string)
```

Comment:
Should we confirm that `[0]` is there before casting to it?

Reply:
as discussed offline - no this is the existing behaviour and it's guaranteed in the schema, but removed in 3.0